pterodactyl.php

```diff
@@ -1251,18 +1251,30 @@ public function getAdminAddFields($package, $vars = null)
         $nest_id = $package->meta->nest_id;
         $egg_id = $package->meta->egg_id;
 
+        $option_groups = [];
+        foreach ($package->option_groups as $option_group) {
+            $option_groups[] = $option_group->id;
+        }
+
+        $package->configurable_options = $this->Form->collapseObjectArray(
+            $this->Record->select(['package_options.id', 'package_options.name'])
+                ->from('package_options')
+                ->innerJoin(
+                    'package_option_group',
+                    'package_option_group.option_id',
+                    '=',
+                    'package_options.id',
+                    false
+                )
+                ->where('package_options.company_id', '=', Configure::get('Blesta.company_id'))
+                ->where('package_option_group.option_group_id', 'IN', $option_groups)
+                ->fetchAll(),
+            'id',
+            'name'
+        );
+
         // Load nest/egg from config option if submitted
         if (isset($vars->configoptions)) {
-            $configurable_options = $this->Form->collapseObjectArray(
-                $this->Record->select(['package_options.id', 'package_options.name'])
-                    ->from('package_options')
-                    ->where('package_options.company_id', '=', Configure::get('Blesta.company_id'))
-                    ->where('package_options.id', 'IN', array_keys($vars->configoptions))
-                    ->fetchAll(),
-                'id',
-                'name'
-            );
-
             if (isset($configurable_options['nest_id'], $vars->configoptions[$configurable_options['nest_id']])) {
```

Comment:
$configurable_options seems to be undefined, maybe you mean $package->configurable_options here?

Reply:
Updated